adagrams/game.py

```diff
@@ -1,11 +1,153 @@
+#We're splitting letters and then throwing them in the pile :-)
+split_box = []
+#Pulling keys + values from the letter pool, multiplying keys by values and putting into the split_box
+for letter, freq in LETTER_POOL.items():
+    letter_multiplied = letter * freq
+    split_box.append(letter_multiplied)
+#new list for all the new letters we've split up!
+letter_pile = [c for letter in split_box for c in letter]
+
+#Wave 1
 def draw_letters():
-    pass
+    hand = []
+    #Drawing 10 letters
+    for i in range(10):
+        char = random.choice(letter_pile)
+        hand.append(char)
+    #When a letter is overused!
+    for letter in hand:
+        if hand.count(letter) > LETTER_POOL[letter]:
+            #Removes letter if it's over the # in pool
+            hand.remove(letter)
+            #Adds a brand new random letter when the length is under 10.. which it should bc lol
+            if len(hand) < 10:
+                hand.append(random.choice(letter_pile))
+    return hand
 
+#wave 2
 def uses_available_letters(word, letter_bank):
-    pass
+    #Ignore cases by making word all upper
+    word = word.upper()
+    #Loop through each letter in word, and checking if it's the letter bank
+    for char in word:
+        if char in letter_bank and word.count(char) == letter_bank.count(char):
+            result = True
+        else:
+            result = False
+    return result
 
 def score_word(word):
-    pass
+    score = 0
+    #Making sure words evaluated are uppercase
+    word = word.upper()
+    #Loop through each letter + add to get each score
+    for letter in word:
+        if letter != letter.isalpha():
+            score += 0
+        if letter == letter in ["A","E","I","O","U","L","N","R","S","T"]:
+            score += 1
+        elif letter == letter in ["D","G"]:
+            score += 2
+        elif letter == letter in ["B","C","M","P"]:
+            score += 3
+        elif letter == letter in ["F","H","V","W","Y"]:
+            score += 4
+        elif letter == "K":
+            score += 5
+        elif letter == letter in ["J","X"]:
+            score += 8
+        elif letter == letter in ["Q","Z"]:
+            score += 10
+        
+    #Lastly, checking to see if that word's length = bonus :)
+    if len(word) >= 7:
+        score += 8
+    return score
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    #Calling scores to make scores for each word in list!
+    scores = [score_word(word) for word in word_list]
+    words = [word for word in word_list]
+    high_scores = {word:score for (word,score) in zip(words,scores)}
+    #make best_word into an empty list!
+    best_word = []
+    #storing the max score + word:
+    max_word = max(high_scores, key=high_scores.get)
+    max_score = max(high_scores.values())
+    #So we'll do it by tie conditions. We set Tie to 0.
+    tie = 0
+#A loop to count how many ties we got
+    for word in high_scores:
```

Comment:
No matter the amount of ties, ultimately we must do the same thing. We need to check if any words are 10 letters long. Or if the word is shorter than the current highest scored word.

Having several for loops that repeat the same tasks over and over is trouble to maintain. Consider one single for loop that will compare each word to find the highest score no matter how long the list.